test/framework/src/main/java/org/elasticsearch/test/TestCluster.java

```diff
@@ -136,52 +235,93 @@ public void assertAfterTest() throws Exception {
+                        // ignore
+                        listener.onResponse(null);
+                    } else if (t instanceof IllegalArgumentException) {
+                        // Happens if `action.destructive_requires_name` is set to true
+                        // which is the case in the CloseIndexDisableCloseAllTests
+                        if ("_all".equals(indices[0])) {
+                            ClusterStateResponse clusterStateResponse = client().admin().cluster().prepareState().get();
+                            ArrayList<String> concreteIndices = new ArrayList<>();
+                            for (IndexMetadata indexMetadata : clusterStateResponse.getState().metadata()) {
+                                concreteIndices.add(indexMetadata.getIndex().getName());
+                            }
+                            if (concreteIndices.isEmpty() == false) {
+                                client().admin()
+                                    .indices()
+                                    .prepareDelete(concreteIndices.toArray(Strings.EMPTY_ARRAY))
+                                    .execute(listener.delegateFailure((l, r) -> {
+                                        assertAcked(r);
+                                        l.onResponse(null);
+                                    }));
+                            }
+                        } else {
+                            // TODO: this is clearly wrong but at least
+                            // org.elasticsearch.xpack.watcher.test.integration.BootStrapTests.testTriggeredWatchLoading depends on this
+                            // quietly passing when it tries to delete an alias instead of its backing indices
+                            listener.onResponse(null);
+                        }
+                    } else {
+                        listener.onFailure(e);
                     }
                 }
-            }
-        }
+            });
     }
 
     /**
      * Removes all templates, except the templates defined in the exclude
      */
-    public void wipeAllTemplates(Set<String> exclude) {
-        if (size() > 0) {
-            GetIndexTemplatesResponse response = client().admin().indices().prepareGetTemplates().get();
-            for (IndexTemplateMetadata indexTemplate : response.getIndexTemplates()) {
-                if (exclude.contains(indexTemplate.getName())) {
-                    continue;
-                }
-                try {
-                    client().admin().indices().prepareDeleteTemplate(indexTemplate.getName()).get();
-                } catch (IndexTemplateMissingException e) {
-                    // ignore
+    private void wipeAllTemplates(Set<String> exclude, ActionListener<Void> listener) {
+        client().admin().indices().prepareGetTemplates().execute(listener.delegateFailure((f, response) -> {
+            try (RefCountingListener subscribableListener = new RefCountingListener(f)) {
```

Comment:
Do we need a separate inner `RefCountingListener` here or could we re-use the outer one?

Reply:
Right, reused the outer one now :)